### PR TITLE
Fetcher AcquireConfig test

### PR DESCRIPTION
**Pull request by alexander871015**
Wrote test for AcquireConfig and changed the filename of hologram_data_source_availability_fetcher.cc since it had an extra before.

### CODE REVIEW

File: hologram_dashboard/fetcher/hologram_config_malformed.config [hunk 1]
```diff
@@ -0,0 +1,63 @@
+data_source_config {
```
Comment (sarahchen-google) [inline hologram_dashboard/fetcher/hologram_config_malformed.config:1]: Move this to hologram_dashboard/fetcher/testdata/

File: hologram_dashboard/fetcher/hologram_data_source_availability_fetcher_test.cc [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+*/
+
+#include "hologram_data_source_availability_fetcher.h"
+#include <filesystem>
+
+namespace wireless_android_play_analytics{
+
+TEST(FetcherTest, AcquireConfigTest) {
+    // Test malformed/invalid path.
+    HologramDataSourceAvailabilityFetcher h;
```
Comment (sarahchen-google) [inline hologram_dashboard/fetcher/hologram_data_source_availability_fetcher_test.cc:24]: Make the variable name meaningful, like hologram_data_source_availability_fetcher.

File: hologram_dashboard/fetcher/hologram_data_source_availability_fetcher_test.cc [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    ASSERT_DEATH(h.AcquireConfig("/abc"), "");
+    ASSERT_DEATH(
+        h.AcquireConfig("fetcher/hologram_config_malformed.config"), "");
+
+    // Test correct path.
+    h.AcquireConfig("fetcher/hologram_config.config");
+    std::ifstream ins("fetcher/hologram_config.config");
+    std::string compare;
```
Comment (sarahchen-google) [inline hologram_dashboard/fetcher/hologram_data_source_availability_fetcher_test.cc:32]: You can define expected_hologram_conifg and filling values inside to do validation.

Don't read file line by line to do comparison.

File: hologram_dashboard/fetcher/hologram_config_malformed.config (hunk 1 above)
Comment (emilyqu) [reply hologram_dashboard/fetcher/hologram_config_malformed.config:1]: +1. And make this file short, with some fake value.

File: hologram_dashboard/fetcher/hologram_data_source_availability_fetcher_test.cc [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            hologram_config_debug.end());
+    ASSERT_EQ(hologram_config_debug, compare);
+}
+
+} // namespace wireless_android_play_analytics
+
+int main(int argc, char *argv[]) {
+    ::testing::InitGoogleTest(&argc, argv);
+    return RUN_ALL_TESTS();
+}
```
Comment (emilyqu) [inline hologram_dashboard/fetcher/hologram_data_source_availability_fetcher_test.cc:54]: ooc, do you need this main function in tests?

Comment (alexander871015) [reply hologram_dashboard/fetcher/hologram_data_source_availability_fetcher_test.cc:48]: yes

File: hologram_dashboard/fetcher/hologram_data_source_availability_fetcher_test.cc [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+namespace wireless_android_play_analytics{
+
+TEST(FetcherTest, AcquireConfigTest) {
+    // Test malformed/invalid path.
+    HologramDataSourceAvailabilityFetcher hologram_fetcher;
+    ASSERT_DEATH(hologram_fetcher.AcquireConfig("/abc"), "");
+    ASSERT_DEATH(hologram_fetcher.AcquireConfig(
+        "fetcher/testdata/hologram_config_malformed.config"), "");
+
```
Comment (emilyqu) [inline hologram_dashboard/fetcher/hologram_data_source_availability_fetcher_test.cc:28]: Split this test into two test cases. 
One for malformed input path, the other for success case.

File: hologram_dashboard/fetcher/BUILD [hunk 6]
```diff
@@ -6,19 +6,30 @@ cc_library(
...
+    data = ["testdata/hologram_config_test.config", 
+            "testdata/hologram_config_malformed.config"]
+)
+
 cc_binary(
     name = "fetcher_main",
     srcs = ["fetcher_main.cc"],
     deps = [
-        "hologram_data_source_availabiliity_fetcher"
+        "hologram_data_source_availability_fetcher"
```
Comment (emilyqu) [inline hologram_dashboard/fetcher/BUILD:33]: super nit: I feel this file name is too long, and no need to include hologram. Can we name it like data_availability_fetcher?

File: hologram_dashboard/fetcher/BUILD [hunk 7]
```diff
@@ -6,19 +6,30 @@ cc_library(
 )
 
 cc_library(
-    name = "hologram_data_source_availabiliity_fetcher",
-    srcs = ["hologram_data_source_availabiliity_fetcher.cc"],
+    name = "hologram_data_source_availability_fetcher",
+    srcs = ["hologram_data_source_availability_fetcher.cc"],
```
Comment (emilyqu) [inline hologram_dashboard/fetcher/BUILD:10]: super nit: I feel this class name is too long, and no need to include hologram. Can we name it like data_availability_fetcher?

File: hologram_dashboard/fetcher/hologram_data_source_availability_fetcher_test.cc [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    HologramConfigSet expected_hologram_config;
+    // Create a proto with expected values
+    HologramConfig* config = expected_hologram_config.add_data_source_config();
+    config->set_source_type(DataSource::APP_TYPE_SOURCE);
+    config->set_kvick_corpus(Corpus::HOLOGRAM_DEFAULT);
+    config = expected_hologram_config.add_data_source_config();
+    config->set_source_type(DataSource::USER_ATTRIBUTE);
+    config->set_kvick_corpus(Corpus::HOLOGRAM_USER_ATTRIBUTE);
+    
+    ASSERT_TRUE(google::protobuf::util::MessageDifferencer::Equals(expected_hologram_config, hologram_fetcher.hologram_configs_));
```
Comment (sarahchen-google) [inline hologram_dashboard/fetcher/hologram_data_source_availability_fetcher_test.cc:44]: nit: keep the length of the line <=80 characters.

File: hologram_dashboard/fetcher/testdata/hologram_config_malformed.config [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+    source_type: APPS_DAILY_DATA_SOURCE
+    kvick_corpus: HOLOGRAM_DEFAULT
+}
+
+data_source_config {
+    source_type: APPS_TITLE_TIER
+    kvick_c: HOLOGRAM_FAKE_PIPE
+}
+
+data_source_config {
```
Comment (sarahchen-google) [inline hologram_dashboard/fetcher/testdata/hologram_config_malformed.config:16]: Add comment here to point out why it is malformed.

File: hologram_dashboard/fetcher/testdata/hologram_config_test.config [hunk 10]
```diff
@@ -0,0 +1,9 @@
+data_source_config {
```
Comment (sarahchen-google) [inline hologram_dashboard/fetcher/testdata/hologram_config_test.config:1]: Change the file name to hologram_config_valid.ascii

File: hologram_dashboard/fetcher/testdata/hologram_config_malformed.config [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
```
Comment (sarahchen-google) [inline hologram_dashboard/fetcher/testdata/hologram_config_malformed.config:1]: For such config files, we usually use .ascii as its extension.

Change the file name to hologram_config_malformed.ascii.

File: hologram_dashboard/fetcher/hologram_data_source_availability_fetcher_test.cc [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+*
+*   Unless required by applicable law or agreed to in writing, software
+*   distributed under the License is distributed on an "AS IS" BASIS,
+*   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*   See the License for the specific language governing permissions and
+*   limitations under the License.
+*/
+
+#include "hologram_data_source_availability_fetcher.h"
+#include <google/protobuf/util/message_differencer.h>
```
Comment (sarahchen-google) [inline hologram_dashboard/fetcher/hologram_data_source_availability_fetcher_test.cc:18]: Blank line between line 17 and 18.

File: hologram_dashboard/fetcher/testdata/hologram_config_test.ascii [hunk 13]
```diff
@@ -0,0 +1,9 @@
+data_source_config {
```
Comment (sarahchen-google) [inline hologram_dashboard/fetcher/testdata/hologram_config_test.ascii:1]: Make the name informative, like hologram_config_valid.ascii.

File: hologram_dashboard/fetcher/hologram_data_source_availability_fetcher_test.cc [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+    HologramConfigSet expected_hologram_config;
+    // Create a proto with expected values
+    HologramConfig* config = expected_hologram_config.add_data_source_config();
+    config->set_source_type(DataSource::APP_TYPE_SOURCE);
+    config->set_kvick_corpus(Corpus::HOLOGRAM_DEFAULT);
+    config = expected_hologram_config.add_data_source_config();
+    config->set_source_type(DataSource::USER_ATTRIBUTE);
+    config->set_kvick_corpus(Corpus::HOLOGRAM_USER_ATTRIBUTE);
+    
+    ASSERT_TRUE(google::protobuf::util::MessageDifferencer::Equals
```
Comment (sarahchen-google) [inline hologram_dashboard/fetcher/hologram_data_source_availability_fetcher_test.cc:45]: EXPECT_TRUE?

**Pull request merged**